### PR TITLE
Add Django CMS

### PR DESCRIPTION
**Pull request by Frost**
...and a bunch of plugins to make it work.

This will be used for building the new version of karspexet.se.

### CODE REVIEW

**Comment by DiTo04** [comment]
@Eleton Jag mergade in den gren vi jobbade på idag i den som berörs av det här PR:et :) Så vi kan jobba direkt här :)

File: karspexet/templates/base.html [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+                {% show_menu 0 100 0 1 %}
+              </ul>
+            </div>
+        </div>
+        {% addtoblock "css" %}
+        <link href="{% static 'fonts.css' %}" rel="stylesheet" type="text/css"/>
+        <link href="{% static 'menu.css' %}" rel="stylesheet" type="text/css"/>
+        {% endaddtoblock %}
+        {% addtoblock "js" %}
+        <script src="https://ajax.googleapis.com/ajax/libs/jquery/3.2.1/jquery.min.js"></script>
```
Comment (maxnordlund) [inline karspexet/templates/base.html]: This is an issue for two major reasons, first it makes the site dependent on googleapis to be up and responsive. Secondly it leads to a [security vulnerability][1]. Please use bower and serve it locally instead.

Same goes for CSS if and when you need to add a library.

[1]: https://www.riskiq.com/blog/labs/jquerycom-malware-attack-puts-privileged-enterprise-it-accounts-at-risk/#.VCG5gBawQeN

File: karspexet/templates/base.html [hunk 2]
```diff
@@ -0,0 +1,91 @@
+{% load menu_tags cms_tags sekizai_tags staticfiles%}
```
Comment (maxnordlund) [inline karspexet/templates/base.html]: What does this one do? I mean shouldn't it be _inside_ the html tag? Also a `<!DOCTYPE html>` would be nice pretty please.

File: static/menu.js [hunk 3]
```diff
@@ -0,0 +1,18 @@
+const navigationBar = $("#navbar");
+
+$("#contentBox").css("top", $("#navbar").css("height")); //Sets the content below the navbar.
+
+document.onscroll = () => {
+    const titleHeight = parseInt($(".titleContainer").css("height"));
```
Comment (maxnordlund) [inline static/menu.js]: Please add a base here: `parseint(.., 10)`, and below.

File: static/menu.js [hunk 4]
```diff
@@ -0,0 +1,18 @@
+const navigationBar = $("#navbar");
+
+$("#contentBox").css("top", $("#navbar").css("height")); //Sets the content below the navbar.
+
+document.onscroll = () => {
```
Comment (maxnordlund) [inline static/menu.js]: This uses ES6 which I think is fine, but that means we need an asset pipeline to transpile it since we will have people using IE.

File: karspexet/templates/base.html [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+              <h1><span class="g">KÅR</span>S<span class="g">P</span>EX<span class="g">ET</span></h1>
+            </div>
+            
+            <div class="horizontal menuContainer">
+              <ul class="menu">
+                {% show_menu 0 100 0 1 %}
+              </ul>
+            </div>
+        </div>
+        {% addtoblock "css" %}
```
Comment (maxnordlund) [inline karspexet/templates/base.html]: It seems a bit convoluted to use the `addtoblock` helper here since it's just ~10 lines below the insertion point, yes?

File: karspexet/templates/base.html (hunk 2 above)
Comment (DiTo04) [inline karspexet/templates/base.html]: It works just as well outside the <html> tag, as it is in the base.html file under /ticket. Yeah should add <!DOCTYPE html>.

File: karspexet/templates/base.html (hunk 1 above)
Comment (DiTo04) [inline karspexet/templates/base.html]: Did not know! Nicely found!

Comment (maxnordlund) [inline karspexet/templates/base.html]: No worries, it's not obvious and until that incident people didn't really consider it dangerous.

File: static/menu.js (hunk 3 above)
Comment (DiTo04) [inline static/menu.js]: haha, this was the thing you talked about yesterday, wasn't it? ^^
Will do!

Comment (maxnordlund) [inline static/menu.js]: Oh yeah, that was it indeed.

File: karspexet/templates/base.html (hunk 2 above)
Comment (Frost) [inline karspexet/templates/base.html]: Can you pleas add a space just before the `%}` as well?

File: karspexet/templates/base.html (hunk 5 above)
Comment (Frost) [inline karspexet/templates/base.html]: I agree.

Comment (DiTo04) [inline karspexet/templates/base.html]: Could be seen as a bit convoluted. I agree, the idea is to make it standard to append styles and scripts to appropriate blocks where they are used. As works goes on more `{% addtoblock %}` should be needed.
Do you believe we are over doing it with blocks and should skip it? :)

File: static/menu.js (hunk 4 above)
Comment (DiTo04) [inline static/menu.js]: Of the to of your head, is there a better way to do it IE friendly?

Comment (maxnordlund) [inline static/menu.js]: Just use a normal function:
```javascript
function() {
    ...
}
```

File: karspexet/templates/base.html (hunk 5 above)
Comment (maxnordlund) [inline karspexet/templates/base.html]: At least in the same template file. If you inherit the layout and want to add a page specific script, then this is the way to go. But here? Not so much, IMO.

File: static/menu.js (hunk 4 above)
Comment (DiTo04) [inline static/menu.js]: Ah, you talking about the `() => {}` part? I thought you where talking about line 3.

File: karspexet/templates/base.html (hunk 2 above)
Comment (DiTo04) [inline karspexet/templates/base.html]: After some experimenting I found that Django templets do not like the `<!DOCTYPE html>` tag. And the styling/padding is changed when it is used...

File: static/menu.js (hunk 4 above)
Comment (maxnordlund) [inline static/menu.js]: Yes, I try to put the line comment on the line it's about when possible.

Comment (Frost) [inline static/menu.js]: We should probably have a look at adding babel soon, so that we can use es6 stuff. That would tidy up the ticket code a bit as well. I just didn't want to add that when the entire project had a total of ~160 lines of js.

Comment (maxnordlund) [inline static/menu.js]: 🤤  I want it so much

**Comment by DiTo04** [comment]
@Frost @maxnordlund: anybody knows why Safari can't show the added font `Pargament`? It downloads the .woff file, but the text is displayed in a default font. The preview of the font in the developer explorer also shows default (read: Wrong!) font...

**Comment by Frost** [comment]
@DiTo04 I have no idea. 

For the record, my personal opinion on web fonts are that they are horrible and should never be used. It might be ok for logos like this, but when used for body text it makes me want to do really horrible things to whoever made the decision to put it there, especially when I'm on a slow network connection. I absolutely _hate_ when text jumps around and/or becomes invisible for short (or long) periods of time when loading a page.

But, let me reiterate: I think it's ok for logos, even if I would have prefered to just use an SVG image instead, or something.

**Comment by DiTo04** [comment]
The git lfs should not have triggered :( Thought I removed the LFS file...

**Comment by Frost** [comment]
I solved the font issue by generating an SVG image, and the menu.js file is now gone since I started using plain CSS to fix the header instead.

I'm not sure whether I really like the idea of a fixed header, but I'll leave those design decisions to someone who is better than I at it.

@maxnordlund Are there still any changes you want to be applied here, or is this good to go for now?

**Comment by DiTo04** [comment]
@Frost Vet du varför det inte genereras någon webpack.stat fil hos circleCI?

File: karspexet/templates/home.html [hunk 6]
```diff
@@ -1,39 +1,43 @@
 {% extends "base.html" %}
 {% load cms_tags  sekizai_tags fontawesome %}
 {% block content %}
```
Comment (DiTo04) [inline karspexet/templates/home.html]: {% clock content %} 
...
{% endblock %}
is a block and thus the indention might be good :)

**Comment by DiTo04** [comment]
@Frost Kan du laga ci/circleci requirement? :)

**Comment by DiTo04** [comment]
@Frost @maxnordlund: A review would be grate :D

**Comment by Frost** [comment]
@DiTo04 Jag kan inte sätta en grön review på denna eftersom det var jag som lade upp den från början. Det är en teknisk begränsning på GitHub. Jag kan däremot gå igenom koden och se hur jag tycker det ser ut, och kan försöka göra det antingen när jag kommer hem ikväll eller imorgon.

File: webpack.config.js [hunk 7]
```diff
@@ -1,5 +1,6 @@
+/* global  __dirname module */
 var path = require("path"),
-    webpack = require("webpack"),
+    //webpack = require("webpack"), //Commented as it is not used, but could be needed later.
```
Comment (Frost) [inline webpack.config.js]: Remove this line. It most probably won't be needed later.

File: assets/css/fonts.css [hunk 8]
```diff
@@ -0,0 +1,56 @@
+/* cyrillic-ext */
```
Comment (Frost) [inline assets/css/fonts.css]: Can someone explain why we would ever need cyrillic fonts?

File: assets/css/karspexet.css [hunk 9]
```diff
@@ -0,0 +1,213 @@
...
+
+ul {
+  padding-left: 0;
+}
+
+.luva {
+  max-width: 5vw;
+  height: auto;
+}
+.list_decorator {
```
Comment (Frost) [inline assets/css/karspexet.css]: This rule is dead code. Remove it.

File: karspexet/settings.py [hunk 10]
```diff
@@ -154,3 +187,81 @@
...
+    },
+    'home.html card_3_image': {
+        'inherit': 'image',
+        'name': "Card 3 Image",
+    },
+    'home.html sponsor_images': {
+        'inherit': 'image',
+        'name': "Sponsor Images",
+    },
+}
```
Comment (Frost) [inline karspexet/settings.py]: This file is missing a line break at end-of-file.

File: karspexet/templates/about.html [hunk 11]
```diff
@@ -0,0 +1,8 @@
+{% extends "base.html" %}
+{% load cms_tags  sekizai_tags fontawesome %}
```
Comment (Frost) [inline karspexet/templates/about.html]: There are double spaces between `cms_tags` and `sekizai_tags`.

**Comment by Frost** [comment]
Efter att koden har levt vidare från den här branchen och den här koden (och det som den levt vidare till) har kört live i produktion under ett verksamhetsår, och för att kunna lättare få hjälp att fixa saker i framtiden, har jag resettat `master` till det som faktiskt kör i produktion. Det innebär att denna PR numera är obsolet, så jag stänger den.

**Comment by maxnordlund** [comment]
@Frost ska grenen också dö, eller ska den flyttas?
